include/supercluster.hpp

```diff
@@ -244,34 +244,24 @@ class Supercluster {
 
         Zoom(Zoom &previous, const double r, const std::uint8_t zoom) {
 
-            // Zoom encoded in the lower 5 bits
-            bool valid_zoom = (((zoom + 1) & 0b11111) == (zoom + 1));
-#ifdef NDEBUG
-            if (!valid_zoom)
-                return;
-#else
-            assert(valid_zoom);
-#endif
+            // The zoom parameter is restricted to [minZoom, maxZoom] by caller
+            assert( ((zoom + 1) & 0b11111) == (zoom + 1) );
 
-            for (std::size_t i = 0; i < previous.clusters.size(); i++) {
+            // Since point index is encoded in the upper 27 bits, clamp the count of clusters
+            const auto previous_clusters_size = std::min(previous.clusters.size(), static_cast<std::vector<Cluster>::size_type>(0x7ffffff));
```

Comment:
> std::vector<Cluster>::size_type

Can this be just `std::size_t`?

Reply:
Good question - is that a safe assumption here? (My C++ is 10 years old and rusty!)

Reply:
I went ahead and merged - please feel free to change if necessary!